test_plans/is_device_copyable.asciidoc

```diff
@@ -0,0 +1,81 @@
+* When doing an inter-device transfer of an object of type `T`, the effect of each eligible copy constructor, move constructor, copy assignment operator, and move assignment operator is the same as a bitwise copy of the object;
+* Type `T` has a public non-deleted destructor;
+* The destructor has no effect when executed on the device.
+
+Define a new class of type `T` with public default constructor, destructor, copy constructor, copy assignment operator and several public members:
+```
+class T {
+  public:
+    T() = default;
+    ~T() = default;
+
+    /* -- common interface members -- */
+
+    int a;
+    float b;
+    char c;
+};
+```
+
+Given that `is_device_copyable<T>::value` can be either `true` or `false`, `integral_constant_t` will be used instead of `std::integral_constant<bool, true>` and `std::integral_constant<bool, false>`.
+
+== Tests
+
+=== Cpp17UnaryTrait requirements.
+
+`is_device_copyable` must meet the https://en.cppreference.com/w/cpp/named_req/UnaryTypeTrait[Cpp17UnaryTrait] requirements:
+
+    * DefaultConstructible and CopyConstructible. Verify that:
+
+        ** `std::is_default_constructible_v<is_device_copyable<T>> == true`;
+        ** `std::is_copy_constructible_v<is_device_copyable<T>> == true`.
+
+    * Takes one template type parameter (additional template parameters are optional and allowed). Verify that: `std::is_constructible_v<is_device_copyable<T>> == true`.
+
+    * Publicly and unambiguously derived from a specialization of integral_constant, known as its base characteristic. Verify that +
+    `std::is_base_of_v<integral_constant_t, is_device_copyable<T>> == true`.
+
+    * The member names of the base characteristic (`integral_constant_t`) are not hidden and are unambiguously available. Verify that:
+
+        ** `integral_constant_t::value_type` and `is_device_copyable<T>::value_type` are the same type;
+        ** `std::integral_constant<bool, true>::type` and `is_device_copyable<T>::type` are the same type;
+        ** Operators `integral_constant_t::value_type() and `is_device_copyable<T>::value_type()` have the same return type;
+        ** `integral_constant_t::operator()` and `is_device_copyable<T>::operator()` return the same value.
+
+=== Specialization for user defined class
+
+Add a specialization for `T`:
+
+`template<> +
+struct is_device_copyable<T> : std::true_type {}` +
+
+* Verify that `is_device_copyable<T>::value == true`;
+
+* Check data transmission:
+
+    ** Initialize a constant instance of class `T` as a reference;
+    ** Create an instance of class `T` on the host by assigning values from the reference object;
+    ** Create an instance of class `T` on the device by assigning values from the reference object;
+    ** Get access to the device object using `accessor` to check data-to-host copying;
+    ** Verify that the object of type `T` in the kernel has the same value that was initialized on the host.
```

Comment:
This new description seems a lot less clear to me.  It seems like you are only testing that `T` is correctly copied from host-to-device, but you are not testing the other way around (device-to-host).  It might be clearer if you wrote some small code snippet showing the basic structure of the test.

Reply:
"Get access to the device object using `accessor` to check data-to-host copying;" isn't this point testing device to host copying? By the way, there is one more typo, device-to-host and instead of data-...

Reply:
I have already written the sample code that is described in this section
```  
  T host_object(reference_instance);
  T device_object_access;
  {
    sycl::buffer<user_def_type_derived, 1> buf(&device_object_access, {1});
    queue.submit([&](sycl::handler& cgh) {
      sycl::accessor<user_def_type_derived, 0, sycl::access_mode::read_write> acc(buf, cgh);
      cgh.single_task<kernel_name>([=] {
        T device_object(reference_instance);
        acc = device_object;
      });
    });
    queue.wait_and_throw();
  }
```

Reply:
Applied.